internal/integrations/rawtemplate.go

```diff
@@ -0,0 +1,241 @@
+
+	rawConfig := rawYAMLConfig{
+		Pipelines:  selectedPipelines,
+		Receivers:  selectedReceivers,
+		Processors: selectedProcessors,
+	}
+	resolver, err := newResolver(rawConfig, params)
+	if err != nil {
+		return nil, err
+	}
+	conf, err := resolver.Resolve(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var config Config
+	if err := conf.Unmarshal(&config); err != nil {
+		return nil, fmt.Errorf("failed to generate effective configuration: %w", err)
+	}
+	return &config, config.Validate()
+}
+
+func newResolver(rawConfig rawYAMLConfig, variables map[string]any) (*confmap.Resolver, error) {
+	settings := confmap.ResolverSettings{
+		URIs: []string{"config:main"},
+		ProviderFactories: []confmap.ProviderFactory{
+			newConfmapProviderFactory(rawConfig),
+			newVariablesProviderFactory(variables),
+		},
+	}
+	return confmap.NewResolver(settings)
+}
+
+type selectableID interface {
+	comparable
+	fmt.Stringer
+}
+
+func selectComponents[C any, ID selectableID](from map[ID]C, selection []ID) (map[ID]C, error) {
+	selected := make(map[ID]C)
+	for _, id := range selection {
+		component, found := from[id]
+		if !found {
+			return nil, fmt.Errorf("component %q not found", id)
+		}
+		selected[id] = component
+	}
+	return selected, nil
+}
+
+func listComponents(pipelines map[pipeline.ID]PipelineConfig, listIDs func(PipelineConfig) []component.ID) []component.ID {
+	var list []component.ID
+	for _, pipeline := range pipelines {
+		list = append(list, listIDs(pipeline)...)
+	}
+	slices.SortFunc(list, func(a, b component.ID) int { return strings.Compare(a.String(), b.String()) })
+	return slices.Compact(list)
+}
+
+// listReceivers lists the IDs of all the receivers found in pipelines
+func listReceivers(pipelines map[pipeline.ID]PipelineConfig) []component.ID {
+	return listComponents(pipelines, func(c PipelineConfig) []component.ID {
+		return []component.ID{*c.Receiver}
+	})
+}
+
+// listReceivers lists the IDs of all the processors found in pipelines
+func listProcessors(pipelines map[pipeline.ID]PipelineConfig) []component.ID {
+	return listComponents(pipelines, func(c PipelineConfig) []component.ID {
+		return c.Processors
+	})
+}
+
+type rawYAMLConfig struct {
+	Receivers  map[component.ID]yaml.Node     `mapstructure:"receivers"`
+	Processors map[component.ID]yaml.Node     `mapstructure:"processors"`
+	Pipelines  map[pipeline.ID]PipelineConfig `mapstructure:"pipelines"`
+}
+
+func (c *rawYAMLConfig) Validate() error {
+	for id, pipeline := range c.Pipelines {
+		if err := pipeline.Validate(); err != nil {
+			return fmt.Errorf("invalid pipeline %q: %w", id, err)
+		}
+
+		if pipeline.Receiver != nil {
+			if _, found := c.Receivers[*pipeline.Receiver]; !found {
+				return fmt.Errorf("receiver %q not defined", pipeline.Receiver.String())
+			}
+		}
+
+		for _, processor := range pipeline.Processors {
+			if _, found := c.Processors[processor]; !found {
+				return fmt.Errorf("processor %q not defined", processor.String())
+			}
+
+		}
+	}
+
```

Comment:
Should there be an error, if `len(c.Pipelines)` is `0`?

Reply:
Good idea, done.